src/ephy-bookmarks-popover.c

```diff
@@ -0,0 +1,530 @@
+  for (iter = g_sequence_get_begin_iter (bookmarks);
+       !g_sequence_iter_is_end (iter);
+       iter = g_sequence_iter_next (iter)) {
+    EphyBookmark *bookmark = g_sequence_get (iter);
+    GtkWidget *row;
+
+    row = create_bookmark_row (bookmark, self);
+    gtk_container_add (GTK_CONTAINER (self->tag_detail_list_box), row);
+  }
+
+  gtk_label_set_label (GTK_LABEL (self->tag_detail_label), tag);
+
+  gtk_stack_set_visible_child_name (GTK_STACK (self->toplevel_stack),
+                                    "tag_detail");
+}
+
+static void
+ephy_bookmarks_popover_list_box_row_activated_cb (EphyBookmarksPopover   *self,
+                                                  GtkListBoxRow          *row,
+                                                  GtkListBox             *box)
+{
+  const char *type;
+  const char *tag;
+
+  g_assert (EPHY_IS_BOOKMARKS_POPOVER (self));
+  g_assert (GTK_IS_LIST_BOX_ROW (row));
+  g_assert (GTK_IS_LIST_BOX (box));
+
+  type = g_object_get_data (G_OBJECT (row), "type");
+  if (g_strcmp0 (type, EPHY_LIST_BOX_ROW_TYPE_BOOKMARK) == 0) {
+    EphyBookmark *bookmark;
+    GActionGroup *action_group;
+    GAction *action;
+    const char *url;
+
+    action_group = gtk_widget_get_action_group (GTK_WIDGET (self->window), "win");
+    action = g_action_map_lookup_action (G_ACTION_MAP (action_group), "open-bookmark");
+
+    bookmark = ephy_bookmark_row_get_bookmark (EPHY_BOOKMARK_ROW (row));
+    url = ephy_bookmark_get_url (bookmark);
+
+    g_action_activate (action, g_variant_new_string (url));
+  } else {
+    tag = g_object_get_data (G_OBJECT (row), "title");
+    ephy_bookmarks_popover_show_tag_detail (self, tag);
+  }
+}
+
+static void
+ephy_bookmarks_popover_set_property (GObject      *object,
+                                     guint         prop_id,
+                                     const GValue *value,
+                                     GParamSpec   *pspec)
+{
+  EphyBookmarksPopover *self = EPHY_BOOKMARKS_POPOVER (object);
+
+  switch (prop_id) {
+    case PROP_WINDOW:
+      self->window = g_value_get_object (value);
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+  }
+}
+
+static void
+ephy_bookmarks_popover_get_property (GObject      *object,
+                                     guint         prop_id,
+                                     GValue       *value,
+                                     GParamSpec   *pspec)
+{
+  switch (prop_id) {
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+  }
+}
+
+static void
+ephy_bookmarks_popover_class_init (EphyBookmarksPopoverClass *klass)
+{
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+  GtkWidgetClass *widget_class = GTK_WIDGET_CLASS (klass);
+
+  object_class->set_property = ephy_bookmarks_popover_set_property;
+  object_class->get_property = ephy_bookmarks_popover_get_property;
```

Comment:
You should just not override get_property at all, since you have no readable properties.